clients/adapter-amqp/src/main/java/org/eclipse/hono/adapter/client/amqp/RequestResponseClient.java

```diff
@@ -0,0 +1,647 @@
+     * @param currentSpan The <em>Opentracing</em> span used to trace the request execution.
+     * @return A future indicating the outcome of the operation.
+     *         The future will be failed with a {@link ServerErrorException} if the request cannot be sent to
+     *         the remote service, e.g. because there is no connection to the service or there are no credits
+     *         available for sending the request or the request timed out.
+     * @throws NullPointerException if any of action, result handler or currentSpan is {@code null}.
+     * @throws IllegalArgumentException if the properties contain any non-primitive typed values.
+     * @see AbstractHonoClient#setApplicationProperties(Message, Map)
+     */
+    public final Future<R> createAndSendRequest(
+            final String action,
+            final Map<String, Object> properties,
+            final Buffer payload,
+            final String contentType,
+            final Function<Message, R> responseMapper,
+            final Span currentSpan) {
+
+        return createAndSendRequest(
+                action,
+                linkTargetAddress,
+                properties,
+                payload,
+                contentType,
+                responseMapper,
+                currentSpan);
+    }
+
+    /**
+     * Creates a request message for a payload and headers and sends it to the peer.
+     * <p>
+     * This method first checks if the sender has any credit left. If not, a failed future is returned immediately.
+     * Otherwise, the request message is sent and a timer is started which fails the returned future,
+     * if no response is received within <em>requestTimeoutMillis</em> milliseconds.
+     * <p>
+     * In case of an error the {@code Tags.HTTP_STATUS} tag of the span is set accordingly.
+     * However, the span is never finished by this method.
+     *
+     * @param action The operation that the request is supposed to trigger/invoke.
+     * @param address The address to send the message to.
+     * @param properties The headers to include in the request message as AMQP application properties.
+     * @param payload The payload to include in the request message as an AMQP Value section.
+     * @param contentType The content type of the payload.
+     * @param responseMapper A function mapping a raw AMQP message to the response type.
+     * @param currentSpan The <em>Opentracing</em> span used to trace the request execution.
+     * @return A future indicating the outcome of the operation.
+     *         The future will be failed with a {@link ServerErrorException} if the request cannot be sent to
+     *         the remote service, e.g. because there is no connection to the service or there are no credits
+     *         available for sending the request or the request timed out.
+     * @throws NullPointerException if any of action, result handler or currentSpan is {@code null}.
```

Comment:
No `result handler` param here. 
It looks like the `responseMapper` param mustn't be null instead here. The corresponding null check is missing below.